python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -650,6 +669,31 @@ def _get_entities_from_payload(self, data):
         else:
             raise RuntimeError("Unable to determine an entity from data: %s" % data)
 
+    def _get_entity_type_whitelist(self):
+        """
+        Gets a set of entity types that are supported by the browser
+        integration. This set is built from a list of constant entity
+        types, plus all entity types that a PublishedFile entity is
+        allowed to link to, per the current site's schema.
+
+        :returns: A set of string entity types.
+        :rtype: set
+        """
+        if self.ENTITY_TYPE_WHITELIST not in self._cache:
+            type_whitelist = constants.BASE_ENTITY_TYPE_WHITELIST
```

Comment:
I know we have fields that are custom for certain projects. Do you know if the `PublishedFile.entity` field can have different list of entities it can refer to between projects? Maybe @nemoDreamer or @francoisjacques could chime in?